scc/light_client/light_client.go

```diff
@@ -64,3 +67,78 @@ func (c *LightClient) Close() {
 func (c *LightClient) Sync() (idx.Block, error) {
 	return c.state.sync(c.provider)
 }
+
+// getAccountProof retrieves and verifies the proof for the given address.
+// It first ensures the client is synchronized before querying for the proof.
+// Returns an error if synchronization fails, if the proof cannot be obtained.
+func (c *LightClient) getAccountProof(address common.Address) (carmen.WitnessProof, error) {
+	// always sync before querying
+	_, err := c.Sync()
```

Comment:
One could argue that it should be the users choice to sync or not to sync. After all, it is offered as an extra function.